### ISSUE TITLE
Questions related to YCB-v dataset

### ISSUE THREAD

**Issue by pipme** [zju-3dv/pvnet#4]
Hi, thanks for sharing the nice code. Several questions related to YCB-v dataset:

1. How can i modify the code to train and evaluate on YCB-v dataset? Would you release the scripts and pretrained models for YCB-v dataset?
2. How many additional sythetic images did you use for training apart from those provided in YCB-v datatet

The link to pretrained models on LINEMOD seems to be broken.

Thank you.

**Comment by pengsida** [comment]
1. To train on YCB-Video dataset, you need additional procedures: download the far too big dataset; adjust the 3D models of object to the canonical frame; adjust the ground truth keypoints to eliminate the ambiguity of symmetry. These are cumbersome procedures, which are far away from the core algorithm.
2. I won't release the scripts that process the YCB-Video dataset, so as the pretrained models, because this requires me to process the YCB-Video dataset again. To be honest, YCB-Video may not be a good dataset for pose estimation.
3. For YCB-Video, 10000 images from blender rendering and 10000 images from fusion.
4. I will release the pretrained models and present a demo on LINEMOD.

**Comment by pipme** [comment]
Thank you for your quick reply.

Why do you think that YCB-Video may not be good for pose estimation? 

For LINEMOD, does the training set contain 183 images from the test sequence?

Thanks.

**Comment by pengsida** [comment]
Reasons
1. YCB-Video is seemingly big, whose images are very similar to each other.
2. The depth of target objects are within a small range, which enables some depth estimation methods to easily overfit the dataset.
3. The noise in the ground truth poses is too strong to be tolerated.

These statements are my personal opinions, which may be wrong.

For LINEMOD, the training set does not overlap with the test sequence. I don't know what you mean by saying "does the training set contain 183 images from the test sequence".

**Comment by pipme** [comment]
Got it. I ask for that because some previous learning-based methods use 15% of the test sequence to generate training images or for training. 

Thanks.

**Comment by pengsida** [comment]
We also use 15% of the images, which is provided in https://github.com/Microsoft/singleshotpose.
I have released the pretrained models and presented a demo.

**Comment by kk1153** [comment]
> 1. To train on YCB-Video dataset, you need additional procedures: download the far too big dataset; adjust the 3D models of object to the canonical frame; adjust the ground truth keypoints to eliminate the ambiguity of symmetry. These are cumbersome procedures, which are far away from the core algorithm.
> 2. I won't release the scripts that process the YCB-Video dataset, so as the pretrained models, because this requires me to process the YCB-Video dataset again. To be honest, YCB-Video may not be a good dataset for pose estimation.
> 3. For YCB-Video, 10000 images from blender rendering and 10000 images from fusion.
> 4. I will release the pretrained models and present a demo on LINEMOD.

 In point 1, what do you mean by adjusting the ground truth keypoints to eliminate the ambiguity of symmetry. Keypoints are generated later from the gt pose information just in the same way as LINEMOD. Can you please correct me?

**Comment by pengsida** [comment]
Look at Sec. 3.3 of the paper "BB8: A Scalable, Accurate, Robust to Partial Occlusion Method for Predicting the 3D Poses of Challenging Objects without Using Depth"

**Comment by kk1153** [comment]
Ok, I understand you mean you choose to not include the ambiguous poses instead avoid ambiguity by including only single side of ambiguity in training data. I don't know if this kind of problem will really be reflected in results as for such symmetric objects, all methods report results using ADD-S/ADI metric (which nullifies it).

**Comment by zeal-github** [comment]
> 1. adjust the 3D models of object to the canonical frame; adjust the ground truth keypoints to eliminate the ambiguity of symmetry

Hi, I have some questions about the YCBV dataset.
1), what do you mean to adjust the objects to the canonical frame. I think in keypoint-based method, we only need to locate 2d key points, and the corresponding relations can be calculated naturally.
2), how to select the key points of symmetric objects in ycbv manually? Can you provide the key points of ycbv objects or give more details.
(I have refer to section 3.3 of BB8. But it seems not keypoints selection method.)

**Comment by pengsida** [comment]
1. The original YCB-v objects are not aligned to xyz axis. So we need to adjust them to the canonical frame for easier rendering.
2. We use FPS algorithm to automatically select keypoints.

**Comment by WilliamKRobert** [comment]
Hi, Do you mind sharing the poses to correct the model points? Many thanks.

**Comment by pengsida** [comment]
What model points? Do you mean adjusting model to a canonical frame?

**Comment by WilliamKRobert** [comment]
Exactly. I was running your code on YCB-Video and was able to get good results for some objects. For can, scissor, etc, the accuracy is actually very low. Would it be the problem of adjusting model to the canonical frame?

**Comment by pengsida** [comment]
Yes. Since can is symmetrical, we need resolve its ambiguity by adjust its ground truth annotation, which additionally requires us to adjust the object to a canonical frame.
This program acheives it: https://github.com/pengsida/configuration/tree/master/ApproxMVBB

**Comment by trevor-taeyeop** [comment]
https://github.com/zju3dv/pvnet/issues/4#issuecomment-471401551
https://github.com/zju3dv/pvnet/issues/69#issuecomment-525272779

**Can I ask the training data detail for YCB?**

You mentioned 
"We had trained multiple classes on YCB. It requires a bigger network and is time consuming."
"For YCB-Video, 10000 images from blender rendering and 10000 images from fusion."

To Train network, 
Does 10000 images from blender means for per objects or for all objects classes?
1. ( real traininng set +**(10000 images for all classes from blender)** +  10000 images from fusion.)
2. ( real traininng set +**(21 classes * 10000 images for each objects from blender)** +  10000 images from fusion.)

**Comment by pengsida** [comment]
Sorry, my partner @liuyuan-pal did this part.
I do not know the concrete training detail.

**Comment by trevor-taeyeop** [comment]
Thanks for the reply. 
@liuyuan-pal 

Also It is helpful, If introduced included data_augmentation part (rotate and crop). 
During Linemod, It is more designed to augmentation one class instance and as i understand, it is hard to extend to multi-class augmentation using current augmentation process.
 
Could you explain how to use augmentation step when training YCB Dataset.

**Comment by liuyuan-pal** [comment]
Hi, Trevor. Sorry for my late reply.
Before writing the paper, we have tried to train one model for multiple classes. But we found it's too time-consuming for training, so we also separately train one model for one class. The results in the paper and the video are produced by the separate model. We apply the same augmentation method as used in LINEMOD actually.

**Comment by pyni** [comment]
Hi~ 
Actually, i am not clear about what 'canonical frame' means. Does it mean the object frame?
Why ambiguity can be solved if the object model is adjusted to the canonical frame?

**Comment by pengsida** [comment]
1. I misuse `canonical frame` here, which means that an object is aligned to xyz axis.
2. The ambiguity cannot be solved but be solved easier, when the YCB-v objects are aligned to xyz axis.

**Comment by pyni** [comment]
Thanks for quick reply. I feel clearer about that.
